lib/v2/screens/profile_screens/settings/interactor/viewmodels/settings_bloc.dart

```diff
@@ -21,5 +21,10 @@ class SettingsBloc extends Bloc<SettingsEvent, SettingsState> {
     if (event is OnNameChanged) {
       yield state.copyWith(profile: state.profile.copyWith(nickname: event.name));
     }
+    if (event is OnCurrencyChanged) {
+      // Just change the state to trigger repaint
+      yield state.copyWith(pageState: PageState.loading);
```

Comment:
this feels like a hack. Why do we need this ?

Reply:
> this feels like a hack. Why do we need this ?

It was a way to trigger the repaint instead to create one more state variable, but anyway this Bloc will still be removed along with its screen.